src/webgpu/api/operation/adapter/requestDevice.spec.ts

```diff
@@ -78,6 +78,90 @@ g.test('invalid')
     t.expect(lost.reason === undefined);
   });
 
+g.test('stale')
+  .desc(
+    `
+    Test that adapter.requestDevice() can successfully return a device once, and once only.
+    - Tests that we can successfully resolve after serial and concurrent rejections.
+    - Tests that consecutive valid attempts only succeeds the first time, returning lost device otherwise.`
+  )
+  .paramsSubcasesOnly(u =>
+    u
+      .combine('initialError', [undefined, 'TypeError', 'OperationError'])
+      .combine('awaitInitialError', [true, false])
+      .combine('awaitSuccess', [true, false])
+      .filter(({ initialError, awaitInitialError }) => {
+        return !(initialError === undefined && awaitInitialError);
```

Comment:
nit: "unless" (inverse of "filter") is slightly easier to read IMO. Done in a commit.